test/functional/legacy/045_folding_spec.lua

```diff
@@ -190,6 +190,17 @@ describe('folding', function()
       0]])
   end)
 
+  it("Regression #6549", function()
+    feed_command('let g:markdown_folding = 1')
+    feed_command('term top')
+    feed_command('e 045_test_data.md')
+
+    feed_command('sleep 2')
```

Comment:
In my testing, the current mode at this moment is Insert Mode. And the cursor would only jump when in Normal Mode.